Tests/TestingApps/PaywallsTester/PaywallsTester/Networking/HTTPClient.swift

```diff
@@ -0,0 +1,229 @@
+            domain: self.domain,
+            headers: self.headers
+        )
+
+        Self.log(.info, request, "Starting")
+
+        let (data, response) = try await self.session.data(for: request)
+        guard let httpResponse = response as? HTTPURLResponse else {
+            throw Error.unrecognizedResponse(response)
+        }
+
+        let statusCode = httpResponse.httpStatusCode
+        Self.log(.info, request, "Finished: \(statusCode.rawValue)")
+
+        guard statusCode.isSuccessfulResponse else {
+            if let error = statusCode.toError() {
+                throw error
+            }
+
+            if let errorResponse = try? JSONDecoder.default.decode(ErrorResponse.self, from: data) {
+                throw Error.errorResponse(statusCode,
+                                          errorResponse.code, 
+                                          errorResponse.originalCode,
+                                          errorResponse.message)
+            } else {
+                throw Error.invalidResponse(httpResponse)
+            }
+        }
+
+        try Task.checkCancellation()
+
+        do {
+            return try await Task.detached(priority: .utility) {
+                try JSONDecoder.default.decode(Response.self, from: data)
+            }.value
+        } catch {
+            throw Error.failedParsingResponse(
+                error,
+                json: String(data: data, encoding: .utf8)
+            )
+        }
+    }
+
+    func removeCookies() {
+        self.session.configuration.httpCookieStorage?.removeCookies(since: .distantPast)
+    }
+
+    private let session: URLSession = {
+        let configuration = URLSessionConfiguration.default
+        configuration.httpShouldSetCookies = true
+        configuration.httpCookieStorage = .sharedCookieStorage(
+            forGroupContainerIdentifier: UserDefaults.sharedAppGroup
+        )
+
+        return URLSession(configuration: configuration)
+    }()
+
+}
+
+extension HTTPClient {
+
+    public enum Error: Swift.Error {
+
+        case unrecognizedResponse(URLResponse)
+        case invalidResponse(HTTPURLResponse)
+        case errorResponse(HTTPStatusCode, BackendErrorCode, Int, String)
+        case failedParsingResponse(Swift.Error, json: String?)
+        case failedBuildingURL
+
+        case notFound
+
+    }
+
+}
+
+// MARK: - Private
+
+private extension HTTPClient {
+
+    static func log(_ level: OSLogType, _ request: URLRequest, _ message: String) {
+        Self.log(
+            level,
+            "\(request.httpMethod ?? "") \(request.url?.absoluteString ?? ""): \(message)"
+        )
+    }
+
+    static func log(_ level: OSLogType, _ message: String) {
+        Self.logger.log(
+            level: level,
+            "\(message)"
+        )
+    }
+    
+    private static let logger = Logger(subsystem: Bundle.main.bundleIdentifier ?? "com.revenuecat.PaywallsTester",
+                                       category: "HTTPClient")
```

Comment:
we can probably extract this one out, right? Not even sure we need a separate logger for it

Reply:
I added a common logging class in this commit, but am not certain this is what you suggesting...?
https://github.com/RevenueCat/purchases-ios/pull/3811/commits/97df002cdb3d81b91365da9329eb1e3531c9b4cf

Reply:
OK yes I think this is right
https://github.com/RevenueCat/purchases-ios/pull/3813#discussion_r1569685698

Reply:
yep!